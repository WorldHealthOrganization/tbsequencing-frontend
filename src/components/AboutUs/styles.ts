import { css } from '@emotion/react';
import { font, fontSizes } from '../typography/fontSizes';
import appColors from '../../styles/colors';
// import bgImage from './assets/topBg.png';

export const topDescriptionSection = css({
  paddingTop: 110,
  paddingLeft: 122,
  paddingBottom: 151,
  // background: `url(${bgImage}) no-repeat`,
  backgroundSize: 'cover',
  display: 'flex',
});

export const header = css({
  fontSize: fontSizes.h0,
  lineHeight: '40px',
});

export const descriptionText = css({
  marginTop: 40,
  marginBottom: 40,
  fontSize: fontSizes.h2,
  lineHeight: '36px',
});

export const topDescriptionSectionLeft = css({
  width: '75%',
});

export const readMoreButtonLabel = css({
  fontFamily: font.bold,
});

export const whoAreWeSection = css({
  marginTop: 64,
  display: 'flex',
  paddingLeft: 54,
  paddingRight: 40,
});

export const whoAreWeSectionLeft = css({

  flexDirection: 'column',
  flexWrap: 'wrap',
});

export const descriptionPrimaryText = css({
  fontSize: 16,
  lineHeight: '32px',
});

export const descriptionSectionHeader = css({
  marginBottom: 24,
});

export const descriptionPaper = css({
  marginRight: 10,
});

export const partnersSection = css({
  marginTop: 50,
  background: 'rgba(221, 239, 249, 0.25)',
  paddingLeft: 54,
  paddingRight: 54,
  paddingTop: 40,
  paddingBottom: 60,
});

export const partnersContainer = css({
  display: 'flex',
  justifyContent: 'center',
  paddingTop: 46,
});

export const footer = css({
  marginTop: 64,
  backgroundColor: appColors.primary.navy,
  paddingTop: 40,
  paddingBottom: 50,
  paddingLeft: 54,
  paddingRight: 64,
});

export const footerHeader = css({
  marginBottom: 24,
  color: appColors.neutral.white,
});

export const contactsContainer = css({
  display: 'flex',
  justifyContent: 'center',
});

export const contactItem = css({
  marginLeft: 126,
});

export const logoSection = css({
  marginTop: 60,
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
});

const legalHeaderBaseStyle = {
  color: appColors.neutral.white,
};

export const copyright = css({
  ...legalHeaderBaseStyle,
});

export const legalNotice = css({
  ...legalHeaderBaseStyle,
  marginLeft: 100,
});

export const legalContainer = css({
  display: 'flex',
});

export const partnerMargin = css({
  marginLeft: 76,
});

export const stepsContainer = css({
  paddingLeft: 54,
  paddingRight: 54,
});

export const stepsWrapper = css({
  display: 'flex',
  justifyContent: 'center',
  marginTop: 24,
});

export const step0 = css({
  top: '8vh',
  right: '17vw',
  width: 312,
});

export const step1 = css({
  top: '1%',
  left: '1%',
  width: 200,
});

export const linkBanner = css({
  color: appColors.neutral.white,
  textDecoration: 'none',
  ':hover': {
    backgroundColor: appColors.secondary.whoBlue,
  },
  display: 'flex',
});

export const linkText = css({
  color: appColors.secondary.whoBlue,
  textDecoration: 'none',
  ':hover': {
    textDecoration: 'underline',
    textDecorationColor: appColors.secondary.whoBlue,
  },
});
