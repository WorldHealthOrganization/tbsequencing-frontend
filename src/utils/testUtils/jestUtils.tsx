import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
