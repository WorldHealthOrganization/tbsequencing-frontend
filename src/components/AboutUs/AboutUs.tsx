/** @jsxImportSource @emotion/react */
import React, { useRef } from 'react';
import * as styles from './styles';
import PrimaryText from '../typography/PrimaryText';
import { AppButton } from '../AppButton/AppButton';
// import { ReactComponent as ScientistImage1 } from './assets/scientistImage1.svg';
// import { ReactComponent as ScientistImage2 } from './assets/scientistImage2.svg';
import { ReactComponent as SeqLogo } from './assets/seqLogo.svg';
import { ReactComponent as FindLogo } from './assets/findLogo.svg';
import { ReactComponent as UniLogo } from './assets/uniLogo.svg';
import { ReactComponent as WHOLogo } from './assets/whoLogo.svg';
import { ReactComponent as WHOLogoBig } from './assets/whoLogoBig.svg';
import { ReactComponent as StepsSvgFull } from './assets/stepsFull.svg';

import H1 from '../typography/H1';
import AppPaper from '../AppPaper';
import Partner from './components/Partner';
import Contact from './components/Contact';
import H3 from '../typography/H3';

const contactLogoProps = {
  width: '200',
  height: '176',
};

const partnersConfig = [
  {
    title: 'Seq&Treat',
    webSite: 'https://www.finddx.org/what-we-do/projects/seqtreat/',
    description: 'Seq&Treat brings next-generation tuberculosis care to underserved communities.',
    image: <SeqLogo {...contactLogoProps} />,
    id: 0,
  },
  {
    title: 'FIND',
    webSite: 'https://www.finddx.org/',
    description: 'We are a global non-profit connecting countries and communities, funders, decisionmakers, healthcare providers and developers.',
    image: <FindLogo {...contactLogoProps} />,
    id: 1,
  },
  {
    title: 'UNITAID',
    webSite: 'https://unitaid.org/#en',
    description: 'Unitaid saves lives by making new health products available and affordable for people in low- and middle-income countries.',
    image: <UniLogo {...contactLogoProps} />,
    id: 2,
  },
  {
    title: 'World Health Organization',
    webSite: 'https://www.who.int/',
    description: 'We champion health and a better future for all.',
    image: <WHOLogo {...contactLogoProps} />,
    id: 3,
  },
];

const contactsConfig = [
  {
    title: 'FIND',
    email: 'sequencing@finddx.org',
    webSite: 'https://www.finddx.org/what-we-do/projects/seqtreat/',
    linkLabel: 'https://www.finddx.org/',
    id: 1,
  },
  {
    title: 'WHO',
    email: 'tbsequencing@who.int',
    webSite: 'https://www.who.int/teams/global-tuberculosis-programme/overview',
    linkLabel: 'https://www.who.int/',
    id: 2,
  },
];

export const AboutUs = () => {
  const stepsRef = useRef<HTMLDivElement>(null!);

  return (
    <div>
      <section css={styles.topDescriptionSection}>
        <div css={styles.topDescriptionSectionLeft}>
          <H1 style={styles.header}>TBKB</H1>
          <PrimaryText style={styles.descriptionText}>
            Welcome to the
            {' '}
            <i>
              Mycobacterium tuberculosis
            </i>
            {' '}
            sequencing and phenotyping knowledgebase of the WHO Global Tuberculosis
            Programme
          </PrimaryText>
          <AppButton
            injectedLabelStyle={styles.readMoreButtonLabel}
            onClick={() => {
              stepsRef.current.scrollIntoView({
                behavior: 'smooth',
              });
            }}
            size="large"
            variant="contained"
          >
            Read More
          </AppButton>
        </div>
      </section>
      <section css={styles.whoAreWeSection}>
        <div css={styles.whoAreWeSectionLeft}>
          <H1 style={styles.descriptionSectionHeader}>Who are we?</H1>
          <AppPaper style={styles.descriptionPaper}>
            <PrimaryText style={styles.descriptionPrimaryText}>
              The WHO Global Tuberculosis Programme works towards the goal of a world free of TB,
              with zero deaths, disease and suffering due to the disease. The team&apos;s mission is
              to lead and guide the global effort to end the TB epidemic through universal access to
              people-centred prevention and care, multisectoral action and innovation.
            </PrimaryText>
          </AppPaper>
        </div>

        {/* <div><ScientistImage1 width="431" height="304" /></div> */}
      </section>
      <section css={styles.whoAreWeSection}>
        {/* <div><ScientistImage2 width="431" height="304" /></div> */}
        <div css={styles.whoAreWeSectionLeft}>
          <H1 style={styles.descriptionSectionHeader}>Mission of the project</H1>
          <AppPaper style={styles.descriptionPaper}>
            <PrimaryText style={styles.descriptionPrimaryText}>
              The tbsequencing portal welcomes any voluntary contributions of combined
              whole genome sequencing and phenotypic drug susceptibility results (pDST) of
              {' '}
              <i>
                M. tuberculosis
              </i>
              {' '}
              clinical isolates. Terms and conditions for the data submitted to the portal are
              available in the data submission page. The tbsequencing portal handles processing
              of the raw sequencing data internally and provides aggregated statistics based on
              genotype calls in its
              different dashboard views. Ultimately, all data collected via this portal will be
              integrated for drug resistance markers identification under the WHO Mutations
              Catalogue initiative. The tbsequencing portal acknowledges the contribution of the
              {' '}
              <a css={styles.linkText} href="https://www.insdc.org/">
                International Nucleotide Sequence Database Collaboration
              </a>
              {' '}
              and uses its public ressources to achieve its goals.
            </PrimaryText>
          </AppPaper>
        </div>
      </section>
      <section ref={stepsRef} css={styles.stepsContainer}>
        <H1>How does it work?</H1>
        <div css={styles.stepsWrapper}>
          <StepsSvgFull width="1340" height="1332" />

        </div>
      </section>
      <section css={styles.partnersSection}>
        <H1 style={styles.descriptionSectionHeader}>Partners and sponsors</H1>
        <div css={styles.partnersContainer}>
          {partnersConfig.map((partnerProps, index) => {
            const marginStyle = index === 0 ? undefined : styles.partnerMargin;

            return <Partner {...partnerProps} key={partnerProps.id} marginStyle={marginStyle} />;
          })}
        </div>
      </section>
      <footer css={styles.footer}>
        <H1 style={styles.footerHeader}>Contacts</H1>
        <div css={styles.contactsContainer}>
          {contactsConfig.map((contact, index) => {
            const style = index === 0 ? undefined : styles.contactItem;

            return (
              <Contact
                {...contact}
                key={contact.id}
                style={style}
              />
            );
          })}
        </div>
        <div css={styles.logoSection}>
          <WHOLogoBig width="228" height="70" />
          <div css={styles.legalContainer}>
            <H3 style={styles.copyright}><a css={styles.linkBanner} href="https://www.who.int/about/policies/terms-of-use">Â© 2024 WHO</a></H3>
            <H3 style={styles.legalNotice}><a css={styles.linkBanner} href="https://www.who.int/about/policies/privacy">Privacy legal notice</a></H3>
          </div>
        </div>

      </footer>
    </div>
  );
};

export default AboutUs;
