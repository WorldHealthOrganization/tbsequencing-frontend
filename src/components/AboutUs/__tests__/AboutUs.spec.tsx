import React from 'react';
import { render, getTreeJSONWithRedux } from '../../../utils/testUtils/jestUtils';
import AboutUs from '../AboutUs';

describe('AboutUs component test', () => {
  test('renders AboutUs component without error', () => {
    render(
      <AboutUs />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <AboutUs />,
    );

    expect(tree).toMatchSnapshot();
  });
});
