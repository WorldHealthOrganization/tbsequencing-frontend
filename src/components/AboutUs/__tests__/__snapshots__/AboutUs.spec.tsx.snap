// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs component test Should render correctly 1`] = `
<div>
  <section
    className="css-681mwz-AboutUs"
  >
    <div
      className="css-12v7j3q-AboutUs"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-880rsd-MuiTypography-root-BaseText"
      >
        TBKB
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-kwxhw4-MuiTypography-root-BaseText"
      >
        Welcome to the 
        <i>
          Mycobacterium tuberculosis
        </i>
         sequencing and phenotyping knowledgebase of the WHO Global Tuberculosis Programme
      </p>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-1sfzpsj-MuiButtonBase-root-MuiButton-root-AppButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-xpvncm-MuiTypography-root-BaseText"
        >
          Read More
        </p>
      </button>
    </div>
  </section>
  <section
    className="css-1u1847n-AboutUs"
  >
    <div
      className="css-1o6wunx-AboutUs"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-euu0lb-MuiTypography-root-BaseText"
      >
        Who are we?
      </p>
      <div
        className="css-xw4omq-AppPaper"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-5nu55l-MuiTypography-root-BaseText"
        >
          The WHO Global Tuberculosis Programme works towards the goal of a world free of TB, with zero deaths, disease and suffering due to the disease. The team's mission is to lead and guide the global effort to end the TB epidemic through universal access to people-centred prevention and care, multisectoral action and innovation.
        </p>
      </div>
    </div>
  </section>
  <section
    className="css-1u1847n-AboutUs"
  >
    <div
      className="css-1o6wunx-AboutUs"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-euu0lb-MuiTypography-root-BaseText"
      >
        Mission of the project
      </p>
      <div
        className="css-xw4omq-AppPaper"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-5nu55l-MuiTypography-root-BaseText"
        >
          The tbsequencing portal welcomes any voluntary contributions of combined whole genome sequencing and phenotypic drug susceptibility results (pDST) of 
          <i>
            M. tuberculosis
          </i>
           clinical isolates. Terms and conditions for the data submitted to the portal are available in the data submission page. The tbsequencing portal handles processing of the raw sequencing data internally and provides aggregated statistics based on genotype calls in its different dashboard views. Ultimately, all data collected via this portal will be integrated for drug resistance markers identification under the WHO Mutations Catalogue initiative. The tbsequencing portal acknowledges the contribution of the
           
          <a
            className="css-1k83lwy-AboutUs"
            href="https://www.insdc.org/"
          >
            International Nucleotide Sequence Database Collaboration
          </a>
           
          and uses its public ressources to achieve its goals.
        </p>
      </div>
    </div>
  </section>
  <section
    className="css-gs2got-AboutUs"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-1i0qxbv-MuiTypography-root-BaseText"
    >
      How does it work?
    </p>
    <div
      className="css-zlvjj5-AboutUs"
    >
      <svg
        height="1332"
        width="1340"
      >
        stepsFull.svg
      </svg>
    </div>
  </section>
  <section
    className="css-1jkrae5-AboutUs"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-euu0lb-MuiTypography-root-BaseText"
    >
      Partners and sponsors
    </p>
    <div
      className="css-1jkr5is-AboutUs"
    >
      <div
        className="css-eqjams-Partner"
      >
        <svg
          height="176"
          width="200"
        >
          seqLogo.svg
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body1 css-kxf6at-MuiTypography-root-BaseText"
        >
          <a
            className="css-fe89j9-Partner"
            href="https://www.finddx.org/what-we-do/projects/seqtreat/"
          >
            Seq&Treat
          </a>
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-nhq2dv-MuiTypography-root-BaseText"
        >
          Seq&Treat brings next-generation tuberculosis care to underserved communities.
        </p>
      </div>
      <div
        className="css-4cd8x7-Partner"
      >
        <svg
          height="176"
          width="200"
        >
          findLogo.svg
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body1 css-kxf6at-MuiTypography-root-BaseText"
        >
          <a
            className="css-fe89j9-Partner"
            href="https://www.finddx.org/"
          >
            FIND
          </a>
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-nhq2dv-MuiTypography-root-BaseText"
        >
          We connect countries, communities, funders, decisionmakers and healthcare providers globally.
        </p>
      </div>
      <div
        className="css-4cd8x7-Partner"
      >
        <svg
          height="176"
          width="200"
        >
          uniLogo.svg
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body1 css-kxf6at-MuiTypography-root-BaseText"
        >
          <a
            className="css-fe89j9-Partner"
            href="https://unitaid.org/#en"
          >
            UNITAID
          </a>
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-nhq2dv-MuiTypography-root-BaseText"
        >
          Unitaid makes health products available and affordable for low- and middle-income countries.
        </p>
      </div>
      <div
        className="css-4cd8x7-Partner"
      >
        <svg
          height="176"
          width="200"
        >
          whoLogo.svg
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body1 css-kxf6at-MuiTypography-root-BaseText"
        >
          <a
            className="css-fe89j9-Partner"
            href="https://www.who.int/"
          >
            World Health Organization
          </a>
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-nhq2dv-MuiTypography-root-BaseText"
        >
          We champion health and a better future for all.
        </p>
      </div>
    </div>
  </section>
  <footer
    className="css-105d0yu-AboutUs"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ux4luz-MuiTypography-root-BaseText"
    >
      Contacts
    </p>
    <div
      className="css-8bsxat-AboutUs"
    >
      <div
        className="css-vtpaov-Contact"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-17mv985-MuiTypography-root-BaseText"
        >
          FIND
        </p>
        <a
          className="css-1tbc68-Contact"
          href="mailto: sequencing@finddx.org"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-zzhh7v-MuiTypography-root-BaseText"
          >
            sequencing@finddx.org
          </p>
        </a>
        <a
          className="css-1tbc68-Contact"
          href="https://www.finddx.org/what-we-do/projects/seqtreat/"
          rel="noreferrer"
          target="_blank"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-zzhh7v-MuiTypography-root-BaseText"
          >
            https://www.finddx.org/
          </p>
        </a>
      </div>
      <div
        className="css-19vlte3-Contact"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-17mv985-MuiTypography-root-BaseText"
        >
          WHO
        </p>
        <a
          className="css-1tbc68-Contact"
          href="mailto: tbsequencing@who.int"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-zzhh7v-MuiTypography-root-BaseText"
          >
            tbsequencing@who.int
          </p>
        </a>
        <a
          className="css-1tbc68-Contact"
          href="https://www.who.int/teams/global-tuberculosis-programme/overview"
          rel="noreferrer"
          target="_blank"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-zzhh7v-MuiTypography-root-BaseText"
          >
            https://www.who.int/
          </p>
        </a>
      </div>
    </div>
    <div
      className="css-r3j2t5-AboutUs"
    >
      <svg
        height="70"
        width="228"
      >
        whoLogoBig.svg
      </svg>
      <div
        className="css-l3vls6-AboutUs"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-gevtu4-MuiTypography-root-BaseText"
        >
          <a
            className="css-luhs0y-AboutUs"
            href="https://www.who.int/about/policies/terms-of-use"
          >
            Â© 2024 WHO
          </a>
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-xl6u2a-MuiTypography-root-BaseText"
        >
          <a
            className="css-luhs0y-AboutUs"
            href="https://www.who.int/about/policies/privacy"
          >
            Privacy legal notice
          </a>
        </p>
      </div>
    </div>
  </footer>
</div>
`;
